lib/publiccloud/gce.pm

```diff
@@ -170,14 +170,17 @@ sub cleanup {
 
     my $region = $self->{provider_client}->{region};
     my $project = $self->{provider_client}->{project_id};
-    my $instance_id = $self->get_terraform_output(".vm_name.value[0]");
+    my $query = ".vm_name.value[0]";
+    my $instance_id = $self->get_terraform_output($query);
     # gce provides full serial log, so extended timeout
-    if (!check_var('PUBLIC_CLOUD_SLES4SAP', 1) && defined($instance_id)) {
-        if ($instance_id =~ /$self->{resource_name}/) {
-            script_run("gcloud compute --project=$project instances get-serial-port-output $instance_id --zone=$region --port=1 > instance_serial.txt", timeout => 180);
+    if (!check_var('PUBLIC_CLOUD_SLES4SAP', 1)) {
+        if ($instance_id) {
+            my $res = script_run("gcloud compute --project=$project instances get-serial-port-output $instance_id --zone=$region --port=1 > instance_serial.txt", timeout => 180);
+            script_run("gcloud compute --project=$project instances get-serial-port-output $instance_id --port=1 -> instance_serial.txt", timeout => 180)
+              if ($res != 0);    # zone::region mismatch, retry without zone
             upload_logs("instance_serial.txt", failok => 1);
         } else {
-            record_info("Warn", "instance_id " . ($instance_id) ? $instance_id : "empty", result => 'fail');
+            record_info("Warn", "instance_id empty", result => 'fail');
```

Comment:
Is this supposed to fail the test or just the step? It will not fail the whole test is that is the intention.

Reply:
At the moment, my intention is never let the routine  stop test run on error, eventually output returned empty; only the GUI test-details will have a non-blocking red box warning. 
This is the current semantic, but open to discussion.